### PR TITLE
Re-structuring numpy indexing routine functions to their respective f…

### PR DESCRIPTION
**Pull request by abdulasiraj**
All numpy indexing routine functions were in a single file but according to the numpy structure & contribution guidelines they should be in their respective files. So, I re-arranged the functions in to their respective files. No new function is implemented!

### CODE REVIEW

File: ivy/functional/frontends/numpy/indexing_routines/generating_index_arrays.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        if sparse:
+            res = res + (idx,)
+        else:
+            res[i] = idx
+    return res
+
+
+# unravel_index
+@to_ivy_arrays_and_back
+def unravel_index(indices, shape, order="C"):
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/numpy/indexing_routines/generating_index_arrays.py:28]: Can we keep this in `indexing_like_operations.py`

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @abdulasiraj 

Thanks for bringing this up🙂
It makes sense, just a few changes are needed then we merge the PR😊

**Comment by abdulasiraj** [comment]
sure

**Comment by abdulasiraj** [comment]
Hi @hirwa-nshuti ,
But in the numpy file structure, unravel is in generating... file. Should i still follow your instruction?

**Comment by hirwa-nshuti** [comment]
> Hi @hirwa-nshuti , But in the numpy file structure, unravel is in generating... file. Should i still follow your instruction?

Well let's leave it just like this only resolve the merge conflict I'll get the PR merged then

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Would you resolve the merge conflicts

**Comment by abdulasiraj** [comment]
> We can't merge the PR with merge conflicts. Mostly it is due to changes added lately after you opened the PR. You'll have to resolve this by updating your branch to the master branch of Ivy🙂 https://lets-unify.ai/docs/ivy/overview/contributing/the_basics.html#managing-your-fork

Sure!

**Comment by abdulasiraj** [comment]
Hi @hirwa-nshuti,
Can we have 5 meeting video chat??

**Comment by hirwa-nshuti** [comment]
You'll have to run `./merge_with_upstream.sh numpy` and fix conflicts then push the changes.
If you face any issue do text me on discord we can continue the discussion there😊

**Review by hirwa-nshuti — APPROVED** [review]
LGTM!
Thanks for the PR🙂

**Pull request merged**